code/datums/dna.dm

```diff
@@ -104,18 +108,47 @@
 	return .
 
 /datum/dna/proc/generate_struc_enzymes()
-	var/list/sorting = new /list(DNA_STRUC_ENZYMES_BLOCKS)
-	var/result = ""
-	for(var/datum/mutation/human/A in GLOB.good_mutations + GLOB.bad_mutations + GLOB.not_good_mutations)
-		if(A.name == RACEMUT && ismonkey(holder))
-			sorting[A.dna_block] = num2hex(A.lowest_value + rand(0, 256 * 6), DNA_BLOCK_SIZE)
-			mutations |= A
-		else
-			sorting[A.dna_block] = random_string(DNA_BLOCK_SIZE, list("0","1","2","3","4","5","6"))
-
-	for(var/B in sorting)
-		result += B
-	return result
+	var/bonus
+	if(species && species.inert_mutation)
+		bonus = get_initialized_mutation(species.inert_mutation)
+	var/list/mutations_temp = GLOB.good_mutations + GLOB.bad_mutations + GLOB.not_good_mutations + bonus
+	mutation_index.Cut()
+	shuffle_inplace(mutations_temp)
+	if(ismonkey(holder))

```

Comment:
This should delegated to mob function/property.

Reply:
It's already /datum/mutation/human